File: MyFirstApp (Java)/app/src/main/java/ar/edu/itba/example/myfirstapp/DisplayMessageActivity.java
package ar.edu.itba.example.myfirstapp;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import ar.edu.itba.example.myfirstapp.databinding.ActivityDisplayMessageBinding;

public class DisplayMessageActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivityDisplayMessageBinding binding = ActivityDisplayMessageBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Get the Intent that started this activity and extract the string
        Intent intent = getIntent();
        String message = intent.getStringExtra(MainActivity.EXTRA_MESSAGE);

        // Set the message as TextView layout text.
        binding.textView.setText(message);
    }
}

File: MyFirstApp (Kotlin)/app/src/main/java/ar/edu/itba/example/myfirstapp/DisplayMessageActivity.kt
package ar.edu.itba.example.myfirstapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import ar.edu.itba.example.myfirstapp.databinding.ActivityDisplayMessageBinding

class DisplayMessageActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = ActivityDisplayMessageBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Get the Intent that started this activity and extract the string
        val message = intent.getStringExtra(EXTRA_MESSAGE)

        // Set the message as TextView layout text.
        binding.textView.text = message
    }
}